vllm/model_executor/layers/fused_moe/pplx_prepare_finalize.py

```diff
@@ -110,9 +121,11 @@ def prepare(
             out_expert_x_scale=expert_x_scale,
             dp_x=a1q,
             dp_x_scale=a1q_scale,
-            indices=rank_topk_ids,
+            indices=rank_topk_ids.to(torch.uint32),
```

Comment:
You shouldn't need this cast anymore. The type of the topk_ids can be controlled by passing torch.uint32 via indices_type to select_experts.